Assets/Scripts/Input/KnightPlayerController.cs

```diff
@@ -36,6 +40,20 @@ protected override void FixedUpdate()
         }
     }
 
+    protected override void OnEnable()
+    {
+        base.OnEnable();
+
+        Combat.Health.DecrementHealthEvent.AddListener(healthUI.decrementKnightHealthUI);
+    }
+
+    protected override void OnDisable()
+    {
+        base.OnDisable();
+
+        Combat.Health.DecrementHealthEvent.RemoveListener(healthUI.decrementKnightHealthUI);
+    }
```

Comment:
There's an error that occurs when a player exits the room and then rejoins it. This can be fixed with the following:

1. Wrap the `AddListener` and `RemoveListener` calls in `photonView.IsMine` if-checks. Same for `DragonPlayerController`.
2. Add a Photon View component to the GameObject in the scene that has the `HealthUi` component.
3. Use RPC calls to sync the decrement methods in `HealthUi.cs` (using the newly added Photon View).